pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/RuntimeASTTransformer.groovy

```diff
@@ -604,6 +605,47 @@ class RuntimeASTTransformer {
         return transformBuildConditionsContainer(original, PostStage.class)
     }
 
+
+    /**
+     * Generates the axes combinations
+     *
+     * @param axes The incoming axes
+     * @param ret The resulting combinations
+     * @return The resulting combinations.
+     */
+    List GeneratePermutations(Map axes){
```

Comment:
`List generatePermutations(Map axes)` - and in practice we can hopefully end up getting it typed better.

Reply:
It has been better typed before but I took it out